src/components/NewSpot/component.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+import { Circle, Map, TileLayer } from "react-leaflet";
+
+import { useDependency } from "../../di";
+
+export interface NewSpotProps {}
+
+export const NewSpot: React.FC<NewSpotProps> = () => {
```

Comment:
This component is a mess and should be refactored!
I guess I can extract the URL parsing for location in the parent component, so that `NewSpot` only cares about the creation form.